src/packet.c

```diff
@@ -132,7 +132,7 @@ void PacketReinit(Packet *p)
             PacketAlertRecycle(p->alerts.alerts, p->alerts.cnt);
         p->alerts.cnt = 0;
     }
-    p->pcap_cnt = 0;
+    p->pcap_v.pcap_cnt = 0;
```

Comment:
this writes into a union that may be used by a different capture method, wonder if that is safe

Reply:
correct, unsafe, but currently no merged capture methods uses it - ndpi plugin uses it but it could be redone to plugin_v member space. So this is a breaking change and only suitable for 9.0.x

Generally, some methods may use it, but some do not, so I would suggest it is a capture method specific. 
Note the problems highlighted in https://github.com/OISF/suricata/pull/13642#issuecomment-3131529803

Reply:
Could you not do `p->pcap_v.pcap_cnt = 0;` only if the current capture method is pcap ?